include/openthread/platform/misc.h

```diff
@@ -61,6 +61,11 @@ extern "C" {
  */
 void otPlatReset(otInstance *aInstance);
 
+#if OPENTHREAD_LINK_PLAT_RESET_VIA_CALLBACK
+typedef void (*otPlatResetCallbackFunction)(otInstance *aInstance);
+void otSetPlatResetCallback(otPlatResetCallbackFunction f);
```

Comment:
How about moving this to [src/posix/platform/include/openthread/openthread-system.h](https://github.com/openthread/openthread/blob/master/src/posix/platform/include/openthread/openthread-system.h)?

If yes, we could rename something like `otSysSetResetCallback()`.

Reply:
When I move it, do I need to keep the OPENTHREAD_LINK_PLAT_RESET_VIA_CALLBACK build option?  Or should I just move it there, but no other platforms will implement it?

Reply:
Oh, wait, never mind.  I see that I'm moving it to a posix-only header file.  Got it.